extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -41,22 +39,37 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
     rawGeoJson?.let {
       ignoreParsedGeoJsonRegistry[sourceId] = false
       onGeoJsonParsedListenerList.add(onGeoJsonParsed)
-      workerHandler.post {
-        val property = it.toPropertyValue()
-        mainHandler.post {
-          geoJsonParsed = true
-          // we set parsed data when sync setter was not called during background work
-          if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
-            setProperty(property, throwRuntimeException = false)
-            onGeoJsonParsedListenerList.forEach {
-              it.onGeoJsonParsed(this)
-            }
+      parsingJob = scope.launch {
+        val property = getPropertyValue(it)
+        geoJsonParsed = true
+        // we set parsed data when sync setter was not called during background work
+        if (ignoreParsedGeoJsonRegistry[sourceId] == false) {
+          setProperty(property, throwRuntimeException = false)
+          onGeoJsonParsedListenerList.forEach {
+            it.onGeoJsonParsed(this@GeoJsonSource)
```

Comment:
should we handle exceptions here? coroutine error handling works in a different way

Reply:
We don't throw an exception but will show the error log in logcat.